oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/authentication/ReactiveOidcIdTokenDecoderFactory.java

```diff
@@ -80,4 +138,28 @@ public final void setJwtValidatorFactory(Function<ClientRegistration, OAuth2Toke
 		Assert.notNull(jwtValidatorFactory, "jwtValidatorFactory cannot be null");
 		this.jwtValidatorFactory = jwtValidatorFactory;
 	}
+
+	/**
+	 * Sets the resolver that provides the expected {@link JwsAlgorithms JWS algorithm}
+	 * used for the signature or MAC on the {@link OidcIdToken ID Token}.
+	 * The default resolves to {@link JwsAlgorithms#RS256 RS256} for all {@link ClientRegistration clients}.
+	 *
+	 * @param jwsAlgorithmResolver the resolver that provides the expected {@link JwsAlgorithms JWS algorithm}
+	 *                             for a specific {@link ClientRegistration client}
+	 */
+	public final void setJwsAlgorithmResolver(Function<ClientRegistration, String> jwsAlgorithmResolver) {
+		Assert.notNull(jwsAlgorithmResolver, "jwsAlgorithmResolver cannot be null");
+		this.jwsAlgorithmResolver = jwsAlgorithmResolver;
+	}
+
+	private static String getJcaAlgorithm(String jwsAlgorithm) {
+		if (JwsAlgorithms.HS256.equals(jwsAlgorithm)) {
+			return "HmacSHA256";
+		} else if (JwsAlgorithms.HS384.equals(jwsAlgorithm)) {
+			return "HmacSHA384";
+		} else if (JwsAlgorithms.HS512.equals(jwsAlgorithm)) {
+			return "HmacSHA512";
+		}
+		return "";
```

Comment:
Seems like we'd want to error here since this would indicate an unexpected state (you are already checking `isHMAC` earlier so to get to this line would indicate an error).